src/main/java/seedu/address/model/ContactTagIntegrationManager.java

```diff
@@ -0,0 +1,188 @@
+package seedu.address.model;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.tag.ReadOnlyTagTree;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.TagTree;
+import seedu.address.model.tag.TagTreeImpl;
+
+/**
+ * Class that controls {@code Tag} relationships and interactions between {@code Tag}s and {@Code Person}s.
+ */
+public class ContactTagIntegrationManager {
+
+    private AddressBook addressBook;
+    private TagTree tagTree;
+
+    /**
+     * Creates a ContactTagIntegrationManager from the given {@code addressBook} and a new tagTree.
+     */
+    public ContactTagIntegrationManager(AddressBook addressBook) {
+        this.addressBook = addressBook;
+        tagTree = new TagTreeImpl();
+    }
+
+    /**
+     * Creates a ContactTagIntegrationManager from the given {@code addressBook} and {@code tagTree}.
+     */
+    public ContactTagIntegrationManager(AddressBook addressBook, ReadOnlyTagTree tagTree) {
+        this(addressBook);
+        this.tagTree.copy(tagTree);
+    }
+
+    public AddressBook getAddressBook() {
+        return addressBook;
+    }
+
+    public TagTree getTagTree() {
+        return tagTree;
+    }
+
+    /**
+     * Returns a {@code set} or {@code person}s that falls under the given {@code tag} and all its sub-tags.
+     *
+     * @param tag {@code tag} to begin search for matching {@code person}s.
+     * @return Set of Persons falling under either the argument {@code tag} or its sub-tags.
```

Comment:
slight inconsistency; first line of documentation says the set returned falls under the given tag _and_ all its sub-tags, but this line says _or_.

Reply:
_or_ would be appropriate, thanks for pointing this out